server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java

```diff
@@ -133,4 +148,39 @@ ClusterConfiguration provideClusterConfiguration(PropertiesProvider propertiesPr
                 .build();
         }
     }
+
+    @Provides
+    @Singleton
+    KeyspacesConfiguration provideKeyspacesConfiguration(PropertiesProvider propertiesProvider) throws ConfigurationException {
+        try {
+            return KeyspacesConfiguration.from(propertiesProvider.getConfiguration(CASSANDRA_FILE_NAME));
+        } catch (FileNotFoundException e) {
+            LOGGER.warn("Could not locate cassandra configuration file. Using default keyspaces configuration instead");
+            return KeyspacesConfiguration.builder().build();
+        }
+    }
+
+    @Provides
+    @Singleton
+    KeyspaceConfiguration provideMainKeyspaceConfiguration(KeyspacesConfiguration keyspacesConfiguration) {
+        return keyspacesConfiguration.mainKeyspaceConfiguration();
+    }
+
+    static class InitializedCluster implements Startable {
+        private final Cluster cluster;
+
+        @Inject
+        private InitializedCluster(Cluster cluster, ClusterConfiguration clusterConfiguration, KeyspacesConfiguration keyspacesConfiguration) {
+            this.cluster = cluster;
+
+            if (clusterConfiguration.shouldCreateKeyspace()) {
+                KeyspaceFactory.createKeyspace(keyspacesConfiguration.mainKeyspaceConfiguration(), cluster);
+                KeyspaceFactory.createKeyspace(keyspacesConfiguration.cacheKeyspaceConfiguration(), cluster);
```

Comment:
what the point having two modules if we create both keyspaces in the same place?